msm/src/serialization/constraints.rs

```diff
@@ -1,7 +1,54 @@
+use ark_ff::Field;
 use kimchi::circuits::expr::{ConstantExpr, Expr};
 
 use crate::columns::Column;
 
+use super::interpreter::InterpreterEnv;
+
 pub struct Env<Fp> {
     pub constraints: Vec<Expr<ConstantExpr<Fp>, Column>>,
 }
+
+impl<F: Field> InterpreterEnv for Env<F> {
```

Comment:
I'm really not sure I understand this design... `InterpreterEnv` is just like .. a bunch of common functions that can be used for building both constraints and witness? Because for building witness one actually needs some values, while for building constraints one needs merely size parameters and no real values. confused :c